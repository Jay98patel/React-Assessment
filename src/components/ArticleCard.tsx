import { useState } from "react";
import "../styles/app.scss";

import Card from "react-bootstrap/Card";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";

const APP_URL = "https://dummy-rest-api.specbee.site";

const ArticleCard = ({ article }: { article: any }) => {
  const imageUrl = APP_URL + article.image;
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <>
      <Card onClick={handleShow} className="cursor-pointer">
        <Card.Img variant="top" src={imageUrl} alt={article.title} />
        <Card.Body>
          <Card.Title
            title={article.title}
            dangerouslySetInnerHTML={{ __html: article.title }}
          ></Card.Title>
          <Card.Text
            title={article.body}
            dangerouslySetInnerHTML={{ __html: article.body }}
          ></Card.Text>
        </Card.Body>
      </Card>
      <Modal closeButton show={show} onHide={handleClose}>
        <Modal.Body>
          <Card className="shadow-none border-0">
            <Card.Img
              className="rounded mb-3"
              variant="top"
              src={imageUrl}
              alt={article.title}
            />
            <Card.Text>
              <div className="d-flex justify-content-between">
                <span>
                  Source : <b>{article.source}</b>
                </span>
                <span>
                  Date : <b>{article.date}</b>
                </span>
              </div>
            </Card.Text>
            <Card.Title
              className="mb-4 mt-3"
              title={article.title}
              dangerouslySetInnerHTML={{ __html: article.title }}
            ></Card.Title>
            <Card.Link href={article.url} target="_blank">
              {article.url}
            </Card.Link>
            <Card.Text
              className="mt-3"
              dangerouslySetInnerHTML={{ __html: article.body }}
            ></Card.Text>
          </Card>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="outline-primary" onClick={handleClose}>
            Close
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};

export default ArticleCard;
