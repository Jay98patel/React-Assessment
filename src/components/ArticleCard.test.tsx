import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ArticleCard from "./ArticleCard";

const article = {
  image: "/path/to/image.jpg",
  title: "Test Article",
  body: "<p>This is the body of the test article.</p>",
  source: "Test Source",
  date: "2024-07-23",
  url: "https://example.com/test-article",
};

test("renders article card with title and body", () => {
  render(<ArticleCard article={article} />);

  const cardTitle = screen.getByRole("heading", { level: 5 });
  const cardBody = screen.getByText("This is the body of the test article.", {
    selector: "p",
  });

  expect(cardTitle).toBeInTheDocument();
  expect(cardBody).toBeInTheDocument();
});

test("opens modal on card click", async () => {
  render(<ArticleCard article={article} />);

  const card = screen.getByRole("button");
  fireEvent.click(card);

  const modal = await screen.findByTestId("article-modal");
  expect(modal).toBeVisible();
});
