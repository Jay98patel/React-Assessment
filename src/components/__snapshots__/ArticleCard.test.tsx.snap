// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders correctly 1`] = `
<div>
  <div
    class="cursor-pointer card"
  >
    <img
      alt="Sample Article"
      class="card-img-top"
      src="https://dummy-rest-api.specbee.site/sample.jpg"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
        title="Sample Article"
      >
        Sample Article
      </div>
      <p
        class="card-text"
        title="This is a sample article body."
      >
        This is a sample article body.
      </p>
    </div>
  </div>
</div>
`;
